### PR TITLE
Restructure and adds CMake

### PR DESCRIPTION
**Pull request by jlaura**
Builds using CMake. Requires that csm be installed in a discoverable place. Updates the README. Restructures the repo to remove the python stuff.

Checked locally with ldd to make sure the linking looked okay.

```bash
ldd libusgscsm.so
	linux-vdso.so.1 (0x00007ffc289bb000)
	libcsmapi.so.3 => /usr/local/lib/libcsmapi.so.3 (0x00007fe419a6f000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fe4196e1000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fe419343000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fe41912b000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fe418d3a000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fe419f5a000)
```

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            src/UsgsAstroLsSensorModel.cpp
+            src/UsgsAstroLsStateData.cpp)
+
+set_target_properties(usgscsm PROPERTIES
+    VERSION ${PROJECT_VERSION}
+    SOVERSION 1
+    PUBLIC_HEADER include/usgscsm/UsgsAstroFramePlugin.h
+    PUBLIC_HEADER include/usgscsm/UsgsAstroFrameSensorModel.h
+    PUBLIC_HEADER include/usgscsm/UsgsAstroLsPlugin.h
+    PUBLIC_HEADER include/usgscsm/UsgsAstroLsSensorModel.h
```
Comment (jessemapel) [inline CMakeLists.txt]: I think we should either make all of our headers public to make testing easier, or make none of them public.

**Comment by twilson271828** [comment]
The 'Checking for working C compiler: /usr/bin/cc' fails when running cmake3.12.0 (the most recent version).    When I run cmake using the 3.9.1 version, I don't get this.

**Comment by jlaura** [comment]
@twilson271828 That is external to this PR and is a CMake / CMake install issue? All of the compiler checks occur prior to hitting any executable code that we have written as far as I know.

**Comment by jlaura** [comment]
Just tested on OS X 10.12.6 with a fresh CMake 3.12.0 install:

```bash
Scanning dependencies of target usgscsm
[ 16%] Building CXX object CMakeFiles/usgscsm.dir/src/UsgsAstroFramePlugin.cpp.o
[ 33%] Building CXX object CMakeFiles/usgscsm.dir/src/UsgsAstroFrameSensorModel.cpp.o
[ 50%] Building CXX object CMakeFiles/usgscsm.dir/src/UsgsAstroLsPlugin.cpp.o
[ 66%] Building CXX object CMakeFiles/usgscsm.dir/src/UsgsAstroLsSensorModel.cpp.o
[ 83%] Building CXX object CMakeFiles/usgscsm.dir/src/UsgsAstroLsStateData.cpp.o
[100%] Linking CXX shared library libusgscsm.dylib
[100%] Built target usgscsm
```

and 

```bash
otool -L libusgscsm.dylib
libusgscsm.dylib:
	@rpath/libusgscsm.1.dylib (compatibility version 1.0.0, current version 0.0.1)
	@rpath/libcsmapi.dylib.3 (compatibility version 0.0.0, current version 0.0.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 307.5.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.60.2)
```

**Comment by jessemapel** [comment]
I'm able to compile with both 3.9 and 3.12

**Pull request merged**